test/e2e/autoscale_test.go

```diff
@@ -0,0 +1,284 @@
+		require.NoError(t, err)
+
+		collectorCount := len(pods.Items)
+		logrus.Infof("Iteration %d found %d pods", i, collectorCount)
+		if collectorCount > maxCollectorCount {
+			maxCollectorCount = collectorCount
+			if quitOnFirstScale && i > 1 {
+				break
+			}
+		}
+
+		// Print events since last iteration.
+		eventList, err := fw.KubeClient.CoreV1().Events(namespace).List(context.Background(), metav1.ListOptions{})
+		require.NoError(t, err)
+		for _, event := range eventList.Items {
+			if event.LastTimestamp.After(lastIterationTimestamp) {
+				logrus.Warnf("Event Type: %s Reason: %s Message: %s Time %v", event.Type, event.Reason, event.Message, event.LastTimestamp)
+			}
+		}
+
+		lastIterationTimestamp = time.Now()
+		time.Sleep(1 * time.Minute)
+	}
+	require.Greater(t, maxCollectorCount, 1, "Collector never scaled")
+}
+
+// Create a simple-prod instance with optional values for autoscaling the collector
+func getSimpleProd(name, namespace, cpuResourceLimit, memoryResourceLimit string, autoscaleCollector bool) *v1.Jaeger {
+	ingressEnabled := true
+	autoscale := true
+	var minReplicas int32 = 1
+	var maxReplicas int32 = 5
+
+	jaeger := &v1.Jaeger{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1.JaegerSpec{
+			Collector: v1.JaegerCollectorSpec{
+				JaegerCommonSpec: v1.JaegerCommonSpec{
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse(cpuResourceLimit),
+							corev1.ResourceMemory: resource.MustParse(memoryResourceLimit),
+						},
+					},
+				},
+			},
+			Ingress: v1.JaegerIngressSpec{
+				Enabled:  &ingressEnabled,
+				Security: v1.IngressSecurityNoneExplicit,
+			},
+
+			Strategy: v1.DeploymentStrategyProduction,
+			Storage: v1.JaegerStorageSpec{
+				Type: "elasticsearch",
+				Options: v1.NewOptions(map[string]interface{}{
+					"es.server-urls": esServerUrls,
+				}),
+			},
+		},
+	}
+
+	if autoscaleCollector {
+		autoscaleCommonSpec := v1.JaegerCommonSpec{
+			Resources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse(cpuResourceLimit),
+					corev1.ResourceMemory: resource.MustParse(memoryResourceLimit),
+				},
+			},
+		}
+		autoscaleCollectorSpec := v1.AutoScaleSpec{
+			Autoscale:   &autoscale,
+			MinReplicas: &minReplicas,
+			MaxReplicas: &maxReplicas,
+		}
+		jaeger.Spec.Collector.JaegerCommonSpec = autoscaleCommonSpec
+		jaeger.Spec.Collector.AutoScaleSpec = autoscaleCollectorSpec
+	}
+
+	return jaeger
+}
+
+func createTracegenDeployment(jaegerInstanceName, namespace, componentName string, testDuration int, replicas int32) *appsv1.Deployment {
+	annotations := make(map[string]string)
+	annotations["sidecar.jaegertracing.io/inject"] = jaegerInstanceName
```

Comment:
The sidecar annotation is being added, but the agent is being explicitly added - including with a hardcoded image.

Can true sidecar injection be used? Allowing the operator to use the appropriate agent image/version?